pkgs/applications/window-managers/miriway/default.nix

```diff
@@ -0,0 +1,78 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, unstableGitUpdater
+, nixosTests
+, cmake
+, pkg-config
+, mir
+, libxkbcommon
+}:
+
+stdenv.mkDerivation rec {
+  pname = "miriway";
+  version = "unstable-2022-12-18";
+
+  src = fetchFromGitHub {
+    owner = "Miriway";
+    repo = "Miriway";
+    rev = "d294c303cb99b7becb0d6686be9a09f0a1f57596";
+    hash = "sha256-H+IZgI1IQxNl5yAygbDKXkyXajGHV/mp9gEqZcp0TeE=";
+  };
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace "\''${CMAKE_INSTALL_PREFIX}/bin" "\''${CMAKE_INSTALL_BINDIR}" \
+      --replace "/usr/share" "\''${CMAKE_INSTALL_DATADIR}" \
+      --replace "/etc" "\''${CMAKE_INSTALL_SYSCONFDIR}"
```

Comment:
`CMAKE_INSTALL_FULL_<dir>` actually, see https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html

Reply:
I disagree, the `FULL` variants doesn't offer a single benefit here.

This replaces paths that are used in `install` invocations. `install` is fully `CMAKE_INSTALL_PREFIX`-aware, so there is nothing to be gained from using the `FULL` variables. See [clang](https://github.com/llvm/llvm-project/blob/e8231eaa154396c5ff3c29314cdca8f2e0118200/clang/CMakeLists.txt#L360-L376) as a notably large CMake user where this works perfectly fine for us (except where we have to [*add* `CMAKE_INSTALL_<dir>` usages that upstream missed with a patch](https://github.com/NixOS/nixpkgs/blob/a100acd7bbf105915b0004427802286c37738fef/pkgs/development/compilers/llvm/8/clang/gnu-install-dirs.patch)), and [CMake's `install` docs](https://cmake.org/cmake/help/latest/command/install.html#installing-files) using regular `CMAKE_INSTALL_<dir>` in their examples.

Explicitly using the `FULL` variants really only matters for i.e. pkg-config file generations where a full path is expected, which this package doesn't do.

Reply:
Thanks for giving the context, this make sense now.